lib/pages/home_screen.dart

```diff
@@ -186,19 +181,27 @@ class _HomeScreenState extends State<HomeScreen> {
             bottom: (isCupertino(context) ? _panelController.panelHeight - 18 : _panelController.panelHeight) + 80,
             child: Consumer(
                 builder: (context, ref, child) {
-                  final followOnLocation = ref.watch(followOnLocationProvider);
+                  final followOnLocation = ref.watch(followOnLocationTargetProvider);
                   return PlatformFAB(
                     onPressed: () async {
                       final status = await ref.read(locationPermissionStatusProvider.notifier).checkPermission();
+                      final followOnLocationTarget = ref.watch(followOnLocationTargetProvider);
                       if (!mounted) return;
                       if (status.permission.authorized) {
-                        ref.read(followOnLocationProvider.notifier)
-                            .update((state) => FollowOnLocationUpdate.always);
+                        switch (followOnLocationTarget) {
+                          case FollowOnLocationTargetState.none:
+                            ref.read(followOnLocationTargetProvider.notifier).update(FollowOnLocationTargetState.currentLocation);
+                          case FollowOnLocationTargetState.currentLocation:
+                            ref.read(followOnLocationTargetProvider.notifier).update(FollowOnLocationTargetState.forestPark);
+                            followOnLocation;
+                          case FollowOnLocationTargetState.forestPark:
+                            ref.read(followOnLocationTargetProvider.notifier).update(FollowOnLocationTargetState.currentLocation);
+                        }
                       } else {
                         showMissingPermissionDialog(
-                            context,
-                            'Location Required',
-                            'Location permission is required to jump to current location'
+                          context,
+                          'Location Required',
+                          'Location permission is required to jump to the current location',
                         );
                       }
                     },
```

Comment:
The child Icon is still comparing the followOnLocationTarget to a FollowOnLocationUpdate, which always evaluates to false. The icon should change to be filled in when its following the current location. Additionally, a third icon should be added to signify having jumped to forest park (some kind of trail icon).